pkg/agent/controllers/agent/agent.go

```diff
@@ -0,0 +1,205 @@
+package agent
```

Comment:
_agent_ appears twice in the import path: _pkg/agent/controllers/agent_. I suggest to shorten to _pkg/agent/controllers_.

Reply:
Problem is, main.go and agent.go can't be in same root. So I need another folder within controllers. We had similar issue with globlanet too, if you recall. I'll refactor this later once have better clarity on how to use admiral API. Was thinking we could use something like.`controllers/datasyncer`.

Reply:
**main.go** goes in _pkg/agent_ as you have it. For the datasyncer part, I believe you'll just instantiate and run the component from admiral, ie you won't need to implement anything.  Either way, if there is another controller or something, it can just reside in _pkg/agent/controllers_ - you don't really need a sub-directory structure under controllers.

Reply:
Good idea. I was thinking we may have multiple controllers, depending on how we're going to distribute the MCS crds so was thinking about multiple dirs, but that can be easily done within a single controller directory too.